### PR TITLE
Add more halstead metrics

### LINKED ISSUE #1883: Add other Halstead metrics
We are using N1, n1, N2 and n2, but others are available in rust-code-analysis now.

```
            |- halstead
            |  |- n1: 6
            |  |- N1: 86
            |  |- n2: 69
            |  |- N2: 147
            |  |- length: 233
            |  |- estimated program length: 436.9979625220206
            |  |- purity ratio: 1.8755277361460112
            |  |- vocabulary: 75
            |  |- volume: 1451.3147548855402
            |  |- difficulty: 6.391304347826087
            |  |- level: 0.1564625850340136
            |  |- effort: 9275.794302964105
            |  |- time: 515.3219057202281
            |  `- bugs: 0.14715651181813164
```

### PR DESCRIPTION
**Pull request by Amotul-raheem**
resolves #1883

### CODE REVIEW

**Comment by Amotul-raheem** [comment]
review required

File: bugbug/commit_features.py [hunk 1]
```diff
@@ -137,95 +137,128 @@ class source_code_file_metrics(object):
     name = "metrics on source code file"
 
     def __call__(self, commit, **kwargs):
+        commit_metrics = commit["metrics"]
```
Comment (suhaibmujahid) [inline bugbug/commit_features.py:140]: This change is unrelated and makes the diff longer. 

Note: If we are going to refactor this, it should be done in a separate PR that is detected for that.

**Review by suhaibmujahid — COMMENTED** [review]
Thank you @Amotul-raheem for your contribution!

Could you please rebase and resolve the conflicts?

File: bugbug/commit_features.py (hunk 1 above)
Comment (Amotul-raheem) [reply bugbug/commit_features.py]: okay, thanks. done

**Comment by Amotul-raheem** [comment]
Addressed the comments.

File: bugbug/repository.py [hunk 2]
```diff
@@ -515,7 +515,6 @@ def get_summary_metrics(obj, metrics_space):
         obj["halstead_N2_min"] = min(obj["halstead_N2_min"], metrics["halstead"]["N2"])
         obj["halstead_n1_min"] = min(obj["halstead_n1_min"], metrics["halstead"]["n1"])
         obj["halstead_N1_min"] = min(obj["halstead_N1_min"], metrics["halstead"]["N1"])
-
```
Comment (suhaibmujahid) [inline bugbug/repository.py:518]: Unrelated change. Could you please revert it?

File: bugbug/commit_features.py [hunk 3]
```diff
@@ -478,26 +433,18 @@ def __call__(self, commit, **kwargs):
                 "halstead_n1_total"
             ],
             "Total function number of operators": merged_metrics["halstead_N1_total"],
-            "Total function length": merged_metrics["halstead_length_total"],
-            "Total calculated estimated function length": merged_metrics[
+            "Total file length": merged_metrics["halstead_length_total"],
+            "Total file estimated program length": merged_metrics[
```
Comment (suhaibmujahid) [inline bugbug/commit_features.py:437]: I think the names are already correct (the same for others). See the name of the function:
https://github.com/mozilla/bugbug/blob/c96c2e2f03a9a03c73f67c3aa1583779b06c42d5/bugbug/commit_features.py#L335

**Review by suhaibmujahid — COMMENTED** [review]
Thank you @Amotul-raheem for addressing the comments. Could you please drop unrelated changes (e.g., key renames)?

In general, if you need to fix or refactor something, it is better to be done separately. You could open an issue to explain the problem/improvement, and then you could open a pull request to resolve the issue.

File: bugbug/repository.py (hunk 2 above)
Comment (Amotul-raheem) [reply bugbug/repository.py:518]: Initially, I was confused looking at this issue, I think it has already been  [done](https://github.com/mozilla/bugbug/pull/2372/files) but not linked to the [open issue](https://github.com/mozilla/bugbug/issues/1883).

That was why I picked it up. I think we can close the PR. Please let me know your thoughts on this.

Comment (suhaibmujahid) [reply bugbug/repository.py:518]: Sorry for that. I will close this PR. Feel free to select another issue. Please let me know if you need any help.
